src/config.rs

```diff
@@ -3787,4 +3788,26 @@ mod tests {
             }
         );
     }
+
+    #[test]
+    fn test_config_template_valid() {
+        let template_config = std::include_str!("../etc/config-template.toml")
+            .lines()
+            .map(|l| l.strip_prefix('#').unwrap_or(l))
+            .join("\n");
+
+        let mut cfg: TiKvConfig = toml::from_str(&template_config).unwrap();
+        cfg.validate().unwrap();
+    }
+
+    #[test]
+    fn test_config_template_matches_default() {
+        let template_config = std::include_str!("../etc/config-template.toml")
+            .lines()
+            .map(|l| l.strip_prefix('#').unwrap_or(l))
+            .join("\n");
+
+        let cfg: TiKvConfig = toml::from_str(&template_config).unwrap();
+        assert_eq!(cfg, TiKvConfig::default());
```

Comment:
This test is currently failing (intentionally) because I'm not sure if/how we should change the values in `config-template.toml`.
Here is a diff (left: `config-template.toml`, right: `TiKvConfig::default()`): https://www.diffchecker.com/Et8GSSHO

Reply:
I'm afraid it's not easy to do the test you want. Some config items' default values are calculated rather than fixed value. For example, readpool.unified.max-thread-count is calculated from the CPU nums.

Reply:
Maybe you can consider set these special items to the same value before doing the assert_eq.

Reply:
Ah, thanks for the comment! Good point! Maybe we could also leave out that second test anyways :)

What would you prefer? Handling special items or leaving out the test?

Reply:
Both is ok to me, and I personally prefer keeping the test case and handling the special items.

Reply:
Cool! I kept the test, but now handle these cases accordingly :) PTAL